### PR TITLE
develop sync master

### PR DESCRIPTION
**Pull request by UDLD**

### CODE REVIEW

**Comment by rainchen** [comment]
@UDLD 
CI failed

**Comment by UDLD** [comment]
@rainchen  ci passed

**Review by rainchen — APPROVED** [review]

**Comment by rainchen** [comment]
@yangzhenlong please review

File: core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java [hunk 1]
```diff
@@ -593,9 +663,13 @@ public QueryResourceResult queryResource(String fromAddr, String permissionAddr)
         List<TypeReference<?>> outputParamters
                 = Collections.singletonList(new TypeReference<DynamicArray<Address>>() {});
         List<Type> resultTypes = CITASystemContract.decodeCallResult(callResult, outputParamters);
+        if (resultTypes.isEmpty())
+            return null;
+        List<String> list = new ArrayList<>();
```
Comment (yangzhenlong) [inline core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java]: when declare an `ArrayList` object should set a default size

File: core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java [hunk 2]
```diff
@@ -593,9 +663,13 @@ public QueryResourceResult queryResource(String fromAddr, String permissionAddr)
...
         ArrayList<Address> results = (ArrayList<Address>) resultTypes.get(0).getValue();
-        List<String> list = new ArrayList<>(results.size());
         for (Address address : results) {
```
Comment (yangzhenlong) [inline core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java]: can be replaced with Java8 stream and lambda expression: 
```
List<String> list = results.stream()
  .filter(address -> 
    (!INTRA_GROUP_USER_MANAGEMENT_ADDR.equalsIgnoreCase(address.getValue()) )
  .map(Address::getValue).collect(Collectors.toList())
```

**Review by yangzhenlong — APPROVED** [review]

**Review by rainchen — APPROVED** [review]

File: core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java (hunk 2 above)
Comment (rainchen) [reply core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java]: @yangzhenlong this is not a feature PR, this is a "sync" PR, no need to review the code quality in this case, FYI.

**Pull request merged**